docs/operation/operation.md

```diff
@@ -470,6 +471,14 @@ using the Redis ring based solution, adds 2 additional roundtrips to
 redis per hit. Make sure you monitor redis closely, because skipper
 will fallback to allow traffic if redis can not be reached.
 
+## Default filters
+
+Kubernetes dataclient supports default filters. You can enable this feature by
+specifying `default-filters-dir`. The defined directory must contain per-service
+filter configurations, with file name following the pattern `${service}.${namespace}`.
```

Comment:
There will be race conditions between the ingress/service/etc being created and the default filter for it being added to the configmap. If the default filters configure something like authentication, this can cause serious issues. It also makes it impossible for an administrator to just setup a bunch of filters for the whole cluster, because they now have to write a cronjob or a dedicated controller to watch the services. Are there actually any use cases for this, or was it just added for flexibility?

Reply:
In general, you are right.

Our use-case behind the change is the following: we want to provide out-of-the-box monitoring, i.e. we will have a facility in place which generates API monitoring filter configurations based on the (published) API specifications. These configs are then persisted as "default filters".

The api-monitoring-controller, which is responsible for generating the default filters, takes into account which services/ingresses are already manually configured and doesn't create default configurations in such cases. No race conditions.

We should extend the documentation with the hint about the possible race conditions.

Reply:
Why not add global default filters as well, which seems to me a more common use case?

Reply:
@szuecs already implemented **global** filters (#1002). This proposal is to have default filters **per service**.

Reply:
We have this on our radar and are very careful about which filters we plan to add here. The current premise is, that each filter chain/route works perfect without this additional filters. If we change this premise, we will make sure that skipper routes will not work unless a default filter configuration is available :smile:.

Reply:
Ah, I've missed the first part. Sorry then!

Reply:
Ah, I've missed the first part. But then why not have the controller update the filters in the service object (or the corresponding ingress) directly, instead of going through the configmap and then trying to map it back to the services in the data client?

Reply:
@aermakov-zalando 

1. This will create a race condition and probably flapping with stackset-controller
2. defaults should work for all dataclients, not only in kubernetes
3. it's more efficient if we read from a file, then to query ingress and get a lot of more or less static data.

Reply:
@szuecs We are going to move the filter configuration to the Service resource (opt-out and custom config). This will be a higher level configuration and will be the basis for `api-monitoring-controller` to generate the effective `apiUsageMonitoring` filter configurations.

Reply:
@szuecs I would say the service resource is the more precise location for API filter configurations, because the setup is related to target service and not some ingress definition. There surely exist other situations - and it is a valid option to implement the same mechanism against ingress resources, but this is not our use case. Anyhow, using the service as reference and key in the configuration makes sure, that we do not have conflicts in situations where one ingress configuration is actually leading to two different service - where we would require two different filters.